meetup/tests.py

```diff
@@ -0,0 +1,156 @@
+from django.test import TestCase
+from meetup.models import Meetup
+from pizzaplace.models import PizzaPlace
+from django.db import IntegrityError, DataError
+from django.core.exceptions import ValidationError
+from meetup.services.meetup_api_lookup_agent import MeetupApiLookupAgent
+from meetup.services.meetup_info_fetch import MeetupInfoFetch
+class TestMeetup(TestCase):
+
+  def test_meetup_is_a_thing(self):
+    meetup = Meetup()
+    self.assertIsInstance(meetup, Meetup)
+
+  def test_valid_meetup_has_name_and_link(self):
+    meetup = Meetup(name='papers we love', meetup_link='http://meetup.com/some-meetup')
+    self.assertEquals(meetup.name, 'papers we love')
+    self.assertEquals(meetup.meetup_link, 'http://meetup.com/some-meetup')
+
+  def test_blank_meetup_link_raises_integrity_error(self):
+    meetup = Meetup(meetup_link='http://meetup.com/some-meetup')
+    self.assertRaises(IntegrityError, meetup.save)
+
+  def test_blank_meetup_name_raises_integrity_error(self):
+    meetup = Meetup(name='papers we love')
+    self.assertRaises(IntegrityError, meetup.save)
+
+  def test_meetup_name_can_not_be_over_500_characters(self):
+    name = 'x' *501
+    meetup = Meetup(name=name, meetup_link='http://meetup.com/some-meetup')
+    self.assertRaises(DataError, meetup.save)
+
+  def test_string_representation_of_meetup(self):
+    m = Meetup(name="Mr. Meetup", meetup_link='http://meetup.com/some-other-meetup')
+    self.assertEquals("Mr. Meetup", str(m))
+
+  def test_meetup_name_is_unique(self):
+    m = Meetup(name="Meetup", meetup_link='http://meetup.com/some-meetup')
+    m.save()
+    n = Meetup(name="Meetup", meetup_link='http://meetup.com/some-other-meetup')
+    self.assertRaises(IntegrityError, n.save)
+
+  def test_meetup_link_is_unique(self):
+    m = Meetup(name="Meetup", meetup_link='http://meetup.com/some-meetup')
+    m.save()
+    n = Meetup(name="Meetup new", meetup_link='http://meetup.com/some-meetup')
+    self.assertRaises(IntegrityError, n.save)
+
+  def test_getting_all_associated_pizzas(self):
+    meetup= Meetup(name="Meeetup1", meetup_link='http://meetup.com/some-meetup')
+    meetup.save()
+    place = PizzaPlace(name="Pete Zazz")
+    place.save()
+    meetup.pizza_places.add(place)
+    self.assertEquals(place, meetup.pizza_places.all()[0])
```

Comment:
The Django queryset API provides a `first()` method, which I think is way more readable than square brackets.